itests/hcatalog-unit/src/test/java/org/apache/hive/hcatalog/listener/TestDbNotificationCleanup.java

```diff
@@ -0,0 +1,299 @@
+    private static IMetaStoreClient msClient;
+    private static IDriver driver;
+    private static MessageDeserializer md;
+    private static HiveConf conf;
+    private int startTime;
+    private long firstEventId;
+    private final String testTempDir = Paths.get(System.getProperty("java.io.tmpdir"), "testDbNotif").toString();
+
+    private static List<String> testsToSkipForReplV1BackwardCompatTesting =
+            new ArrayList<>(Arrays.asList("cleanupNotifs", "cleanupNotificationWithError","skipCleanedUpEvents"));
+    // Make sure we skip backward-compat checking for those tests that don't generate events
+
+    private static ReplicationV1CompatRule bcompat = null;
+
+    @Rule
+    public TestRule replV1BackwardCompatibleRule = bcompat;
+
+    static {
+        try {
+            md = MessageFactory.getInstance(JSONMessageEncoder.FORMAT).getDeserializer();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+
+
+    @SuppressWarnings("rawtypes")
+    @BeforeClass
+    public static void connectToMetastore() throws Exception {
+        conf = new HiveConf();
+
+        MetastoreConf.setVar(conf,MetastoreConf.ConfVars.TRANSACTIONAL_EVENT_LISTENERS,
+                "org.apache.hive.hcatalog.listener.DbNotificationListener");
+        conf.setBoolVar(HiveConf.ConfVars.HIVE_IN_TEST_REPL, true);
+        conf.setBoolVar(HiveConf.ConfVars.HIVE_SUPPORT_CONCURRENCY, false);
+        conf.setBoolVar(HiveConf.ConfVars.FIRE_EVENTS_FOR_DML, true);
+        conf.setBoolVar(HiveConf.ConfVars.REPLCMENABLED, false);
+        conf.setVar(HiveConf.ConfVars.METASTORE_RAW_STORE_IMPL, DummyRawStoreFailEvent.class.getName());
+        MetastoreConf.setTimeVar(conf, MetastoreConf.ConfVars.EVENT_DB_LISTENER_CLEAN_INTERVAL, CLEANUP_SLEEP_TIME, TimeUnit.SECONDS);
+        MetastoreConf.setVar(conf, MetastoreConf.ConfVars.EVENT_MESSAGE_FACTORY, JSONMessageEncoder.class.getName());
+        MetastoreConf.setTimeVar(conf, MetastoreConf.ConfVars.EVENT_DB_LISTENER_TTL, EVENTS_TTL, TimeUnit.SECONDS);
+        MetastoreConf.setTimeVar(conf, EVENT_DB_LISTENER_CLEAN_STARTUP_WAIT_INTERVAL, 20, TimeUnit.SECONDS);
+
+        conf.setVar(HiveConf.ConfVars.HIVE_AUTHORIZATION_MANAGER,
+                "org.apache.hadoop.hive.ql.security.authorization.plugin.sqlstd.SQLStdHiveAuthorizerFactory");
+        SessionState.start(new CliSessionState(conf));
+
+        msClient = new HiveMetaStoreClient(conf);
+        driver = DriverFactory.newDriver(conf);
+        md = JSONMessageEncoder.getInstance().getDeserializer();
+
+        bcompat = new ReplicationV1CompatRule(msClient, conf, testsToSkipForReplV1BackwardCompatTesting);
+    }
+
+    @Before
+    public void setup() throws Exception {
+        long now = System.currentTimeMillis() / 1000;
+        startTime = 0;
+        if (now > Integer.MAX_VALUE) {
+            fail("Bummer, time has fallen over the edge");
```

Comment:
`startTime` is not used at all. I wonder, are the other parts of this method are necessary?

Reply:
fixed

Reply:
thx